stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/GiphyMediaAttachmentViewStyle.kt

```diff
@@ -0,0 +1,100 @@
+package io.getstream.chat.android.ui.message.list.adapter.view
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import android.widget.ImageView
+import androidx.annotation.ColorInt
+import androidx.annotation.DimenRes
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.common.extensions.internal.dpToPx
+import io.getstream.chat.android.ui.common.extensions.internal.getColorCompat
+import io.getstream.chat.android.ui.common.extensions.internal.getDrawableCompat
+import io.getstream.chat.android.ui.common.extensions.internal.getEnum
+import io.getstream.chat.android.ui.common.extensions.internal.use
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.GiphyMediaAttachmentView
+
+/**
+ * Sets the style for [io.getstream.chat.android.ui.message.list.adapter.view.internal.GiphyMediaAttachmentView] by obtaining
+ * styled attributes.
+ *
+ * @param progressIcon Displayed while the Giphy is Loading.
+ * @param giphyIcon Displays the Giphy logo over the Giphy image.
+ * @param placeholderIcon Displayed while the Giphy is Loading.
+ * @param imageBackgroundColor Sets the background colour for the Giphy container.
+ * @param giphyConstantSizeEnabled If enabled sets a constant size on all Giphy containers.
+ * @param giphyMaxHeight Sets the maximum height a Giphy container is allowed to have.
+ * @param giphyType Sets the Giphy type which directly affects image quality.
+ * @param scaleType Sets the scaling type for loading the image.
+ * E.g. 'centerCrop', 'fitCenter', etc...
+ */
+public class GiphyMediaAttachmentViewStyle(
+    public val progressIcon: Drawable,
+    public val giphyIcon: Drawable,
+    public val placeholderIcon: Drawable,
+    @ColorInt public val imageBackgroundColor: Int,
+    public val giphyConstantSizeEnabled: Boolean,
+    @DimenRes public val giphyMaxHeight: Int,
+    public val giphyType: GiphyMediaAttachmentView.Companion.GiphyInfoType,
+    public val scaleType: ImageView.ScaleType,
+) {
+    internal companion object {
+        private const val DEFAULT_HEIGHT_DP = 200
+
+        /**
+         * Fetches styled attributes and returns them wrapped inside of [GiphyMediaAttachmentViewStyle].
+         */
+        operator fun invoke(context: Context, attrs: AttributeSet?): GiphyMediaAttachmentViewStyle {
+            context.obtainStyledAttributes(
+                attrs,
+                R.styleable.GiphyMediaAttachmentView,
+                R.attr.streamUiMessageListMediaAttachmentStyle,
+                R.style.StreamUi_MessageList_GiphyMediaAttachment
+            ).use { attributes ->
+                val progressIcon =
+                    attributes.getDrawable(R.styleable.GiphyMediaAttachmentView_streamUiGiphyMediaAttachmentProgressIcon)
+                        ?: context.getDrawableCompat(R.drawable.stream_ui_rotating_indeterminate_progress_gradient)!!
```

Comment:
I would prefer not forcing '!!' here but this is a code style that we've widely adopted when fetching styled attributes so I did not want to break it.